packages/themes/src/components/datepicker.ts

```diff
@@ -0,0 +1,148 @@
+import { YoruStyleProperties } from '@yoru-ui/core';
+
+const opacity = keyframes`
+  0% {
+   opacity: 0;
+  }
+  50% {
+    opacity: 1;
+  }
+  100% {
+    opacity: 1;
+}
+`;
+
+const baseStyle: YoruStyleProperties = {
+  position: 'relative',
+  width: '100%',
+  '& .yoru-click-listener': {
+    transition: 'all .3s',
+    border: '1px solid',
+    borderColor: '#EDEDED',
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    bottom: 0,
+    top: 0,
+    cursor: 'pointer',
+    borderRadius: 4,
+    _hover: {
+      borderColor: 'sky.500',
+    },
+  },
+  '& .date-picker-overlay': {
+    position: 'absolute',
+    backgroundColor: 'white',
+    marginTop: '0.3em',
+    width: '100%',
+    minWidth: '260px',
+    maxWidth: '280px',
+    zIndex: 100,
+    boxShadow: '0 3px 10px rgb(0 0 0 / 0.2)',
+    padding: '1em 0 0.4em 0',
+    animation: 'all 0.2s',
+    borderRadius: '0.25rem',
+    '& .date-picker--header': {
+      marginBottom: '1em',
+      display: 'flex',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      padding: '0 0.5em',
+      '& .prev-month, .next-month': {
+        padding: '0 5px',
+        border: '1px solid',
+        borderColor: 'gray.200',
+        height: '20px',
+      },
+      '& p': {
+        margin: 0,
+        '& span': {
+          cursor: 'pointer',
+          _hover: {
+            textDecoration: 'underline',
+          },
+        },
+      },
+    },
+    '& .date-picker--weekdays': {
+      display: 'grid',
+      gridTemplateColumns: 'repeat(7, minmax(0, 1fr))',
+      placeItems: 'center',
+      fontSize: '0.75em',
+      fontWeight: 'bold',
+      margin: '0.5em 0',
+    },
+    '& .date-picker--days': {
+      display: 'grid',
+      gridTemplateColumns: 'repeat(7, minmax(0, 1fr))',
+      placeItems: 'center',
+      justifyContent: 'center',
+      alignItems: 'center',
+      '& > span.inactive': {
+        '& > span': {
+          fontSize: '0.7em',
+          color: 'gray.300',
+          cursor: 'not-allowed',
+        },
+      },
+      '& > span:not(.inactive)': {
+        cursor: 'pointer',
+        margin: '0.2em',
+        _hover: {
+          color: 'white',
+          animation: `${opacity} 200ms ease-in-out`,
+          '& > span': {
+            cursor: 'pointer',
+            backgroundColor: 'pink.500',
+            borderRadius: '50%',
+          },
+        },
+        '& > span': {
+          display: 'flex',
+          justifyContent: 'center',
+          alignItems: 'center',
+          cursor: 'pointer',
+          width: '24px',
+          height: '24px',
+          fontSize: '0.7em',
+          textAlign: 'center',
+        },
+        '&.selected-date': {
+          margin: '0.2em',
+          '& > span': {
+            border: '1px solid',
+            borderColor: 'pink.500',
+            borderRadius: '50%',
+          },
+        },
+      },
+    },
+    '& .date-picker--months, .date-picker--years': {
```

Comment:
Writing a className like this in my opinion is good, but there are still some mistakes in its implementation. If you write like this why don't you just use BEM 101 CSS?


so if you already have className ``date-picker-overlay`` and you have 3 children of element;

<img width="785" alt="image" src="https://user-images.githubusercontent.com/48151470/205469609-70159d84-884a-4c0e-a545-6e499ab4bc7c.png">


you just can write like this:
<img width="442" alt="image" src="https://user-images.githubusercontent.com/48151470/205469687-667c72ca-181e-42c5-b8db-643dcc2b317a.png">

and the CSS implementation could be like this:

```
.date-picker-overlay {
 "&__header": {
  //css header
 },
  "&__weekdays": {
  //css weekdays
 },
 "&__days": {
  // css days
 }
}
```

Reply:
and for the modifier you can write like this:

for example you have 2 style variant date picker
1. bordered
2. less border

``<div class="yoru-date-picker yoru-date-picker--bordered">``
or
``<div class="yoru-date-picker yoru-date-picker--lessBordered">``

you can write like this:

```
.yoru-date-picker : {
 "&--bordered": {
   // CSS bordered
 }
 "&--lessBordered":  {
   // CSS less bordered
 }
}
```

Reply:
to be changed in the next commit, good review 👍🏻

Reply:
fixed in https://github.com/yoru-ui/yoru-ui/pull/40/commits/4fe627f30c95de3bf3b92cce00cabef00571b140